docs/edgeql/operators/json.rst

```diff
@@ -46,8 +46,23 @@ particular JSON type can be cast back into that scalar:
 - JSON *null* is special since it can be cast into an ``{}`` of any type
 - JSON *array* can be cast into any valid EdgeDB array, so it must be
   homogeneous, and must not contain *null*
-- JSON *object* can be cast into a valid EdgeDB named tuple, but it must not
-  contain *null*
+
+It is possible to cast :eql:type:`uuid` and :ref:`date and time types
+<ref_datamodel_scalars_datetime>` to JSON, the result will be a JSON
+*string* representing the original value. This means that it is also
+possible to cast a JSON *string* back into these types. The string
+value has to be properly formatted (much like in case of a
+:eql:type:`str` value being cast) or else the cast will raise an
+exception.
+
+*Unnamed* :eql:type:`tuple` is converted into a JSON *array* when cast
+into :eql:type:`json`. Whereas *named* :eql:type:`tuple` is converted
+into a JSON *object*. These casts are not reversible, i.e. it is not
+possible to cast a JSON value directly into a :eql:type:`tuple`.
+
+
+JSON Element Reference
```

Comment:
I don't understand what "JSON Element" or "JSON Element Reference" mean.  I'd split this section in two sections: "Accessing JSON Array Elements" and "Accessing JSON Object Fields".

Reply:
It was an attempt to mirror similar sub-titles from the "Array" section. I should propbably rename these sub-sections for the arrays too.